src/workspace-tree/bazel_workspace_tree_provider.ts

```diff
@@ -22,49 +22,38 @@ import { BazelWorkspaceFolderTreeItem } from "./bazel_workspace_folder_tree_item
  * interface.
  */
 export class BazelWorkspaceTreeProvider
-  implements vscode.TreeDataProvider<IBazelTreeItem>
+  implements vscode.TreeDataProvider<IBazelTreeItem>, vscode.Disposable
 {
-  public onDidChangeTreeData: vscode.Event<IBazelTreeItem | void>;
-
   /** Fired when BUILD files change in the workspace. */
-  private onDidChangeTreeDataEmitter =
+  private readonly _onDidChangeTreeData =
```

Comment:
Distinguishing between the emitter and the even only by the `_` is a bit implicit. Let's rather make this more explicit in the variable name

```suggestion
  private readonly onDidChangeTreeDataEmitter =
```

Reply:
Edited all occurrences in this file.